tianshou/data/buffer/base.py

```diff
@@ -72,38 +82,44 @@ def __init__(
         assert stack_num > 0, "stack_num should be greater than 0"
         self.stack_num = stack_num
         self._indices = np.arange(size)
+        # TODO: remove double negation and different name
         self._save_obs_next = not ignore_obs_next
         self._save_only_last_obs = save_only_last_obs
         self._sample_avail = sample_avail
         self._meta = cast(RolloutBatchProtocol, Batch())
-        self._ep_rew: float | np.ndarray
-        self.reset()
+
+        # Keep in sync with reset!
+        self.last_index = np.array([0])
```

Comment:
Why is the index `self.last_index` an array and `self._insertion_idx` is not?

Reply:
Last index is part of the public interface and it exists as a member of the vectorized buffer, where it's an array with multiple entries. Insertion index is private and does not exist in the vectorized buffer (and doesn't need to exist there, or to have the same type)